src/figuresClasses.ts

```diff
@@ -1,19 +1,95 @@
-export interface Figure {
+enum Color {
+  red = 'red',
+  green = 'green',
+  blue = 'blue',
+}
 
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
 
+function isValidLength(...args: number[]): boolean {
+  return args.some((a: number) => a <= 0);
 }
 
-export class Circle {
+function round(param: number): number {
+  return Math.floor(100 * param) / 100;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = Shape.triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const maxLength: number = Math.max(a, b, c);
+    const validLength: boolean = maxLength >= a + b + c - maxLength;
 
+    if (isValidLength(a, b, c)
+     || validLength) {
```

Comment:
In my opinion, validLength it's bad name, becase in this contex this is invalidLength or another name.